plugins/packages/polygon/snapshot.go

```diff
@@ -284,7 +351,7 @@ func (service *PolygonBorService) GetSnapshot(ctx context.Context, blockNrOrHash
 			snap, err = service.getKeyFrame(origin, int64(degree))
 			if err != nil {
 				log.Error("GetSnapshot keyframe case error")
-				return nil, err
+				return nil, nil
```

Comment:
Is returning a nil error in this case correct behavior?

Reply:
My feeling is that the public method should not return errors as it could potentially expose our database structure to users. I went through and standardized error handling in this file. Private methods return errors as well as logs them, whereas the public method only logs them.